### PR TITLE
Created Authentication flow

### PR DESCRIPTION
**Pull request by kjais1720**
## Overview
- Created userSlice and user services to manage and update user state.
- configured axios to set the baseURL and authorization header
- Created Login and Signup components and pages
- Configured routing to public and private pages
- Created dummy pages for routes other than login & signup
## Feature
- User can now login and signup on the app.
- If the user is not logged in and opens the home page of the app, they will be redirected to the landing page but if they were already logged in, they'll be redirected to the timeline page
- On the landing page user can click a link to visit the login page and login or signup there

## Preview
- Login page

image

- Signup page
image

### CODE REVIEW

File: src/components/Illustration.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+export function LoginIllustration(){
```
Comment (ttahm3d) [inline src/components/Illustration.jsx]: This could have been better put in an svg file

**Review by ttahm3d — APPROVED** [review]
Looks good. Have only one suggestion to move illlustration to SVG file.

File: src/components/Illustration.jsx (hunk 1 above)
Comment (kjais1720) [reply src/components/Illustration.jsx]: Thanks for pointing it out, I'll move it to an sbg file

File: src/components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    <div className="min-h-screen bg-primary p-4 md:p-16">
+      <div className="rounded-2xl bg-light-100 dark:bg-dark-100 lg:flex">
+        <div className=" md:rounded-r-0 rounded-2xl bg-light-200 p-8 dark:bg-dark-200 md:rounded-l-2xl xl:max-w-screen-sm">
+          {/* <div className="py-12 bg-light-200 dark:bg-dark-200 flex justify-center lg:justify-start lg:px-12">
+            <div className="cursor-pointer flex items-center">
+              <div className="text-2xl text-primary tracking-wide ml-2 font-semibold">
+                Social Gully
+              </div>
+            </div>
+          </div> */}
```
Comment (rahulcs754) [inline src/components/Login.jsx]: i think you should remove this

**Review by rahulcs754 — APPROVED** [review]
The code looks clean and good

File: src/components/Login.jsx (hunk 2 above)
Comment (kjais1720) [reply src/components/Login.jsx]: Thanks for pointing it out, I have removed it in the latest commit

**Pull request merged**